android/cli/commands/_build.js

```diff
@@ -2510,6 +2510,113 @@ AndroidBuilder.prototype.copyResources = function copyResources(next) {
 		}
 	}, this);
 
+	// genereate missing app icons
+	tasks.push(function (cb) {
+		this.currentBuildManifestFiles = this.currentBuildManifestFiles || {};
```

Comment:
Part of the problem is `this.currentBuildManifestFiles` is something you created. It doesn't exist before this PR. It is never persisted to disk. So, it has no effect.

The way I see it, you can either remove `this.currentBuildManifestFiles` in which case the missing icons will be generated every build -or- hack something to track this state.

Simply checking if the destination image exists would be a step in the right direction. Combine that with checking the mtime on the source DefaultIcon*.png with the mtime of the generated images and you should have an OK mechanism.